### PR TITLE
fix: combine actions

### PR DESCRIPTION
**Pull request by dristpunk**
Closes BES-94

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,15 +22,15 @@ jobs:
...
+          cache: 'yarn'
 
       - name: Install dependencies
         run: yarn --frozen-lockfile --network-concurrency 1
 
       - name: Precompile using 0.8.14 and via-ir=false
         run: yarn build
 
-      - name: "Create env file"
+      - name: 'Create env file'
```
Comment (gas1cent) [inline .github/workflows/ci.yml:33]:
```suggestion
      - name: Create env file
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  forge:
+  tests:
     name: Run Unit and E2E Tests
```
Comment (0xGorilla) [inline .github/workflows/ci.yml:11]: Let's separate the unit and integration tests, they should not depend on one another

**Review by gas1cent — DISMISSED** [review]

**Review by excaliborr — APPROVED** [review]
lgtm!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -21,16 +21,16 @@ jobs:
...
+          cache: 'yarn'
 
       - name: Install dependencies
         run: yarn --frozen-lockfile --network-concurrency 1
 
       - name: Precompile using 0.8.14 and via-ir=false
         run: yarn build
 
-      - name: "Create env file"
+      - name: Create env file
```
Comment (0xOneTony) [inline .github/workflows/ci.yml:33]: Why are unit tests need env variables? I dont see the tests using it

**Review by gas1cent — APPROVED** [review]

**Review by 0xOneTony — APPROVED** [review]

**Review by excaliborr — APPROVED** [review]
lgtm!

**Pull request merged**